dbms/src/Common/RadixSort.h

```diff
@@ -43,61 +43,36 @@ struct RadixSortMallocAllocator
-  *  that the order relation over the keys will match the order relation over the obtained unsigned numbers.
-  * For floats this conversion does the following:
-  *  if the signed bit is set, it flips all other bits.
-  * In this case, NaN-s are bigger than all normal numbers.
-  */
 template <typename KeyBits>
-struct RadixSortFloatTransform
+struct RadixSortIdentityTransform
 {
-    /// Is it worth writing the result in memory, or is it better to do calculation every time again?
-    static constexpr bool transform_is_simple = false;
-
-    static KeyBits forward(KeyBits x)
-    {
-        return x ^ ((-(x >> (sizeof(KeyBits) * 8 - 1))) | (KeyBits(1) << (sizeof(KeyBits) * 8 - 1)));
-    }
+    static constexpr bool transform_is_simple = true;
 
-    static KeyBits backward(KeyBits x)
-    {
-        return x ^ (((x >> (sizeof(KeyBits) * 8 - 1)) - 1) | (KeyBits(1) << (sizeof(KeyBits) * 8 - 1)));
-    }
+    static KeyBits forward(KeyBits x)     { return x; }
+    static KeyBits backward(KeyBits x)    { return x; }
 };
 
 
+
 template <typename TElement>
-struct RadixSortFloatTraits
+struct RadixSortUIntTraits
 {
-    using Element = TElement;     /// The type of the element. It can be a structure with a key and some other payload. Or just a key.
-    using Key = Element;          /// The key to sort by.
-    using CountType = uint32_t;   /// Type for calculating histograms. In the case of a known small number of elements, it can be less than size_t.
-
-    /// The type to which the key is transformed to do bit operations. This UInt is the same size as the key.
-    using KeyBits = std::conditional_t<sizeof(Key) == 8, uint64_t, uint32_t>;
-
-    static constexpr size_t PART_SIZE_BITS = 8;    /// With what pieces of the key, in bits, to do one pass - reshuffle of the array.
+    using Element = TElement;
+    using Key = Element;
+    using CountType = uint32_t;
+    using KeyBits = Key;
 
-    /// Converting a key into KeyBits is such that the order relation over the key corresponds to the order relation over KeyBits.
-    using Transform = RadixSortFloatTransform<KeyBits>;
+    static constexpr size_t PART_SIZE_BITS = 8;
 
-    /// An object with the functions allocate and deallocate.
-    /// Can be used, for example, to allocate memory for a temporary array on the stack.
-    /// To do this, the allocator itself is created on the stack.
+    using Transform = RadixSortIdentityTransform<KeyBits>;
     using Allocator = RadixSortMallocAllocator;
 
-    /// The function to get the key from an array element.
     static Key & extractKey(Element & elem) { return elem; }
-};
-
-template <typename KeyBits>
-struct RadixSortIdentityTransform
-{
-    static constexpr bool transform_is_simple = true;
 
-    static KeyBits forward(KeyBits x)     { return x; }
-    static KeyBits backward(KeyBits x)    { return x; }
+    static bool compare(TElement x, TElement y)
```

Comment:
It is named `less`, not `compare`.

Reply:
Why we compare Elements, should we compare Keys instead?

Reply:
Yes, that's a mistake.